rasa/train_chunks.py

```diff
@@ -0,0 +1,578 @@
+            output,
+            force_training,
+            fixed_model_name,
+            core_additional_arguments,
+            nlu_additional_arguments,
+        ),
+        loop,
+    )
+
+
+async def train_in_chunks_async(
+    domain: Union[Domain, Text],
+    config: Text,
+    training_files: Optional[Union[Text, List[Text]]],
+    output: Text = DEFAULT_MODELS_PATH,
+    force_training: bool = False,
+    fixed_model_name: Optional[Text] = None,
+    core_additional_arguments: Optional[Dict] = None,
+    nlu_additional_arguments: Optional[Dict] = None,
+) -> Optional[Text]:
+    """Trains a Rasa model using smaller chunks (Core and NLU).
+
+    Args:
+        domain: Path to the domain file.
+        config: Path to the config for Core and NLU.
+        training_files: Paths to the training data for Core and NLU.
+        output: Output path.
+        force_training: If `True` retrain model even if data has not changed.
+        fixed_model_name: Name of model to be stored.
+        core_additional_arguments: Additional training parameters for core training.
+        nlu_additional_arguments: Additional training parameters forwarded to training
+                                  method of each NLU component.
+
+    Returns:
+        Path of the trained model archive.
+    """
+    file_importer = TrainingDataImporter.load_from_config(
+        config, domain, training_files
+    )
+    with ExitStack() as stack:
+        train_path = stack.enter_context(TempDirectoryPath(tempfile.mkdtemp()))
+
+        domain = await file_importer.get_domain()
+
+        if domain.is_empty():
+            return await handle_domain_if_not_exists(
+                file_importer, output, fixed_model_name
+            )
+
+        return await _train_in_chunks_async_internal(
+            file_importer,
+            train_path,
+            output,
+            force_training,
+            fixed_model_name,
+            core_additional_arguments=core_additional_arguments,
+            nlu_additional_arguments=nlu_additional_arguments,
+        )
+
+
+async def handle_domain_if_not_exists(
+    file_importer: TrainingDataImporter,
+    output: Text,
+    fixed_model_name: Optional[Text] = None,
+):
+    """Trains only NLU as the domain is not valid or does not exist.
+
+    Args:
+        file_importer: `TrainingDataImporter` which supplies the training data.
+        output: Output path.
+        fixed_model_name: Name of model to be stored.
+
+    Returns:
+        Path of the trained model archive.
+    """
+    nlu_model_only = await _train_nlu_in_chunks_with_validated_data(
+        file_importer, output=output, fixed_model_name=fixed_model_name
+    )
+    print_warning(
+        "Core training was skipped because no valid domain file was found. "
+        "Only an NLU-model was created. Please specify a valid domain using "
+        "the '--domain' argument or check if the provided domain file exists."
+    )
+    return nlu_model_only
```

Comment:
this is exactly the same code as for the regular training besides the function name. Can we de-duplicate this?